docs/_09_development/_develop_guide/_develop_guide/_develop_new_pkg.rst

```diff
@@ -0,0 +1,378 @@
+.. _development_guide_new_pkg:
+
+------------------------
+Developing a New Package
+------------------------
+
+.. _development_guide_new_pkg_architecture:
+
+Architecture of a New Package
+=============================
+
+Structure of a C++ Package
+--------------------------
+
+Every new C++ package should have the following architecture:
+
+.. code-block::
+
+    /package_name
+        /include
+            /package_name
+                ### header files ###
+                node_name(.hpp)
```

Comment:
Something like:
    /package_name
        /include
            /package_name
                header_files(.hpp)
                package_name.hpp
...
/src
   implementation_files(.cpp)
   package_name.cpp
   package_name_node.cpp
...
/tests
   package_name_gtest.cpp
   package_name_test.cpp

Note that gtest must end with _gtest.cpp and normal test executables with _test.cpp (see tests/CMakeLists.txt)

Reply:
is it compulsory for every package to have a header file with the same name as the package? same for implementation and main files. Wouldn´t it be 'node_name.hpp', for example?

Reply:
The standar way of aerostack2 it this. There are some exceptions, but mainly we try to use the package name for simple packages that have only one node